### PR TITLE
chore: GroupListPicker - Bug fixes and adding the possibility to add a default value

### PR DESCRIPTION
**Pull request by mathiasbronner**
Signed-off-by: Mathias Bronner <mathiasb@spotify.com>

## Hey, I just made a Pull Request!

Bug fixes:
- Vertical size jump on text entry
- Left align for text
- Selecting a value closes the popup 
- Autofocus on the popup when opening

Addition: 
- New prop `defaultValue` to adding a default value

_glp

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mathiasbronner** [comment]
> @mathiasbronner you have two changeset files now, feel free to delete one of them 😄

fixed! thx. Didn't know how it worked with the resulting changeset files. now i do 🤓

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks like a lot of unrelated commits snuck in here 😅 You might want to try to address it here or open a new PR 🙏

**Comment by mathiasbronner** [comment]
> Thanks for the PR! Looks like a lot of unrelated commits snuck in here 😅 You might want to try to address it here or open a new PR 🙏

@jhaals Thx, fixed!

File: plugins/org-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (jhaals) [inline plugins/org-react/src/components/GroupListPicker/GroupListPicker.tsx:114]: This is only rendered when the TextField is opened right? Thinking about cases where we might have things like search in the same page.

File: plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx
Comment (jhaals) [inline plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx:30]: Could you explain what this does, I'm not sure what `& p` represent in the context of the picker 😅

**Review by jhaals — COMMENTED** [review]
Nice looks better now! A few questions but otherwise looking good to me :)

File: plugins/org-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (mathiasbronner) [reply plugins/org-react/src/components/GroupListPicker/GroupListPicker.tsx:114]: yes, exactly. The a11y concern for this is that the user does not understand the context switch, but in this case it's an active action to actually clicking or pressing the GroupListPicker button. Inputting or scrolling down is the next logical thing to do

File: plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx
Comment (mathiasbronner) [reply plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx:30]: as this styles the button, the `& p` corresponds to the text part of the button.

Comment (freben) [reply plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx:30]: Alright. Since this is a MUI component and its internal implementation might in theory change over time, we should do that kind of styling in their preferred way instead.

https://v4.mui.com/api/button/

So, render the button as `<Button classes={{ label: classes.buttonLabel }}` and make a `buttonLabel` style here

Comment (mathiasbronner) [reply plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx:30]: Good suggestion, but it does not really fix the problem without pointing to the child paragraf element as the label in the classes object corresponds with the parent span element that wraps the icon (in a span), the text (in a paragraf) and the chevron icon (in a span). See screenshot below:

Screenshot 2022-11-21 at 10 05 24
Screenshot 2022-11-21 at 09 59 20

Do you want me to make change suggested in the screenshot or not @freben ?

Comment (freben) [reply plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx:30]: Alright, then style the `Typography` which generates the `p` instead, right?

Comment (mathiasbronner) [reply plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx]: Yes, i could just style the `Typography`, don't know why i missed that =). 
Fixed

**Review by freben — APPROVED** [review]
Alright!

**Review by jhaals — APPROVED** [review]
🌟

**Comment by mathiasbronner** [comment]
I don't have write access to the repo, so I can't merge the PR 🐼. Is this something one of you could help me with @jhaals or @freben 🤞?

**Comment by jhaals** [comment]
We got you covered, thanks for doing this @mathiasbronner!

**Comment by mathiasbronner** [comment]
thx @jhaals! Where can I see when it's getting released 🤓?

**Pull request merged**
